service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -968,23 +984,31 @@ public TpsPaoUpdateResult updatePolicy(
             userRequest.getEmail());
       }
 
-      HashSet<String> addedGroups =
-          TpsUtilities.getAddedGroups(paoBeforeUpdate, dryRun.getResultingPao());
-      if (!addedGroups.isEmpty()) {
-        logger.info(
-            "Group policies have changed, adding additional groups to auth domain in Sam for workspace {}",
-            workspaceUuid);
-        Rethrow.onInterrupted(
-            () ->
-                samService.addGroupsToAuthDomain(
-                    userRequest,
-                    SamConstants.SamResource.WORKSPACE,
-                    workspaceUuid.toString(),
-                    addedGroups.stream().toList()),
-            "updateAuthDomains");
-      }
+      patchWorkspaceAuthDomain(workspaceUuid, userRequest, paoBeforeUpdate, dryRun);
 
       return result;
     }
   }
+
+  private void patchWorkspaceAuthDomain(
```

Comment:
Do auth domain groups ever get removed (like in the case of having to undo a step)?